src/redux/types.ts

```diff
@@ -41,3 +41,28 @@ export interface GetProductsWithImagesProps {
   images: ImageProps[];
   error?: boolean;
 }
+
+export interface SearchProductsProps {
+  page: number;
+  size: number;
+  body: BodySearchProducts;
+}
+
+export type BodySearchProducts =
+  | {
+      category: string;
+    }
+  | {
+      subcategory: Subcategory;
+    };
```

Comment:
`value: Subcategory.JACKETS`
typeof Subcategory.JACKETS === 'string'
I'm not sure that we need this type, because it's string, please check it